cli/internal/cmd/init.go

```diff
@@ -262,6 +265,8 @@ func (d *initDoer) logGRPCStateChanges(ctx context.Context, wg *sync.WaitGroup,
 		}
 		if state == connectivity.Ready {
 			d.log.Debugf("Connection ready")
+			d.spinner.Stop()
+			d.spinner.Start("Initializing cluster ", false)
```

Comment:
I think we should either prevent this spinner change from happening more than once in this func or prevent multiple init calls all together (see my other comment).

@daniel-weisse @3u13r any thoughts on preventing multiple init calls after the atls conn was established at least once?

Reply:
Dropping from an established connection to connecting should result in a failure imo.
We need to refactor the whole init logic anyway. Currently, a failed attestation will result in a warning about this being a non recoverable error (which is not true).

Reply:
Should we make a double `connectivity.Ready` a failure for now and discuss a larger refactor to init in an upcoming dev sync?

Because I would also +1 for a refactor just because of #1403 for which proper fixes also might include a refactor to init (from the bootstrapper side) to remove or better mitigate the pesky timeout logic.

Reply:
```shell
2023-03-14T08:39:56+01:00       DEBUG   cmd/init.go:246 Created protoClient
2023-03-14T08:39:56+01:00       DEBUG   cmd/init.go:259 Connection state started as CONNECTING
2023-03-14T08:39:56+01:00       DEBUG   cloudcmd/validators.go:170      Validating attestation document
2023-03-14T08:39:57+01:00       DEBUG   cloudcmd/validators.go:170      Successfully validated attestation document
2023-03-14T08:39:57+01:00       DEBUG   cmd/init.go:261 Connection state changed to CONNECTING
2023-03-14T08:39:57+01:00       DEBUG   cmd/init.go:264 Connection ready
// Init actually starts
```
Connection changes to `connecting` twice, so we should definitely not cancel here. But aborting with unretriable error after "Connection read" seems ok to me.

Reply:
Alright, I gave it a try with channels in the new commit. Check it out,  this makes things definitely more complicated though 😄